src/parallel.cpp

```diff
@@ -564,11 +565,12 @@ void HandleBlockMessageThread(CNode *pfrom, const string &strCommand, const CBlo
         int nTotalThinBlocksInFlight = 0;
         {
             LOCK2(cs_vNodes, pfrom->cs_mapthinblocksinflight);
+
             // Erase this thinblock from the tracking map now that we're done with it.
-            if (pfrom->mapThinBlocksInFlight.erase(inv.hash))
```

Comment:
the thinblock in flight is erased now within and at the end of ClearThinBlockData(pfrom, inv.hash) ( the next line down) This keeps the ordering correct since we want to make sure to only erase the thinblock in flight after the thinblock data is deleted.